src/modm/platform/clock/sam_pmc/clockgen_impl.hpp.in

```diff
@@ -28,6 +31,48 @@ ClockGen::enableMainInternal(MainInternalFreq freq)
 	while (!(PMC->PMC_SR & PMC_SR_MOSCRCS));
 }
 
+%% if target.family == "e7x/s7x/v7x"
+template<uint32_t frequency>
+bool
+ClockGen::enableMainExternalCrystal(std::chrono::microseconds startupTime)
+{
+	// disable external clock, clear startup time bits
+	PMC->CKGR_MOR = (PMC->CKGR_MOR | CKGR_MOR_KEY_PASSWD) &
+		~(CKGR_MOR_MOSCXTBY | CKGR_MOR_MOSCXTST_Msk);
+	// wait for external clock to shutdown
+	while (PMC->PMC_SR & PMC_SR_MOSCXTS);
+
+	// convert microseconds to startup time register value
+	// register unit is SCLK cycles (32768 Hz) multiplied by 8
+	const auto cycles = startupTime.count() * (32'768ll / 8) / 1'000'000;
```

Comment:
Just for clarity: 1us input = 1 * (32768 / 8) /  1e6 = 1 * 4096 / 1e6 = 0.004096 cycles? 1ms = 4 cycles?

Reply:
I guess so